### ISSUE TITLE
Whitespace issues again

### ISSUE THREAD

**Issue by Pharap** [Dreamer2345/Forge-Level-Editor#41]
I've noticed that mixed line endings are still happening from time to time,
but I've solved part of what's causing the issue.

It seems that when Visual Studio automatically applies user formatting it does so using the system default line endings instead of following the file's convention.
So basically if you try to insert a space normally it will behave,
but if you do anything that triggers the code formatting system then you'll end up with native line endings.

After learning this, I'm tempted to switch the endings to CRLF because we're all using Windows computers.
(Except for @filmote, but you almost always use git anyway so the `.gitattributes` rule will stop that introducing LFs.)

**Comment by Pharap** [comment]
> core.autocrlf

I'm aware of the existance of that setting, but I haven't actually been using Git at all,
I've been comitting directly via GitHub, which seems to not have a way to handle changing line endings.

---

It wouldn't be a problem if Visual Studio 2019 were smart enough to not change line endings,
but unfortunately while it's smart enough to detect line endings,
it's not smart enough to prevent mixed line endings occurring,
which is really daft because VSCode and other editors are smart enough to do that.

I'll look into something else when I get chance. For now I just have to be careful when comitting.
If I have some time to kill I might write a script to do line ending conversion.

I'm half tempted to suggest we just switch to CRLF anyway because we all have a Windows computer (though @filmote flits between Windows and Mac).

**Comment by Pharap** [comment]
I did a bit of digging and I might have found a solution.
I found [this SO answer](https://stackoverflow.com/a/54121839) which says that Visual Studio supports [.editorconfig files](https://editorconfig.org/), and sure enough [Visual Studio has a default template for it](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2017).

Importantly, it _should_ force Visual Studio to behave itself,
so I'm going to make one that just handles indentation and spaces.
(Apparently it also handles style issues, so we can have fun fighting over that later. `:P`)
